src/cdf/client.ts

```diff
@@ -118,6 +124,17 @@ export async function getLabelsForTarget(
       const timeseries = await getTimeseries({ items: tsIds }, connector, false);
       return timeseries.map((ts) => getLabelWithInjectedProps(labelSrc, ts));
     }
+    case Tab.CogniteTimeSeriesSearch: {
+      // For CogniteTimeSeriesSearch, we use the name from the selected timeseries
+      // since instanceId format doesn't work with regular timeseries endpoints
+      if (target.cogniteTimeSeriesSearchQuery?.selectedTimeseries?.name) {
+        const selectedName = target.cogniteTimeSeriesSearchQuery.selectedTimeseries.name;
+        // If we have a custom label with variables, we can't inject properties from regular timeseries metadata
+        // since this is a DMS instance, so we'll use the selected name as fallback
+        return labelSrc && !labelContainsVariableProps(labelSrc) ? [labelSrc] : [selectedName];
```

Comment:
we need to call instance by id. we should not save the name in the JSON, it can change